src/Data/Bin/Bin0.agda

```diff
@@ -0,0 +1,686 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Arithmetic of binary represented natural numbers. Initial part.
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.Bin.Bin0 where
```

Comment:
Please follow standard library conventions. This module should be split up into Data.Bin and Data.Bin.Properties

Reply:
What about `Data.Bin.Unsigned`? This would leave us enough space to also have `Data.Bin.Signed`
for a binary version of `Data.Integer`.

Reply:
> What about Data.Bin.Unsigned? This would leave us enough space to also have 
> Data.Bin.Signed
> for a binary version of Data.Integer.

I have BInteger/ to propose -- if you do not care for size. 
It has  the type  ℤb  for binary represented Integer. See  BInteger/  in BFLib-0.1.
And I propose to remain with Bin and BInteger, and no "unsigned" word will be needed.
ℤb is provided with a DecIntegralRing instance. So that Fraction over ℤb will be rational numbers in binary coding.

Reply:
I like `Data.Bin.Unsigned` and `Data.Bin.Signed`, they're definitely more standard terminology than `BInteger`. Please leave `Data.Bin.Signed/BInteger` to another PR though.

Reply:
1) This is an _algebraic_ part of the library. In textbooks on algebra, they do not use the notions of "signed" and "unsigned". They use the notions of Natural number and Integer number,
"nonnegative" and "negative".
No one calls nonnegative integer "unsigned natural". No one calls Integer "signed natural".
There is no need for this terminology.
If one talks about _representation_, one says "binary represented natural (Bin)",
 "binary represented integer" (say ``ℤb`` or ``BInt``).

2) Nether do I see such in lib-1.0 (in ``Data/Integer/Base, in Data/Bin/*``).
3) See the declaration  of a binary represented integer  ``ℤb``  in BInteger/BI0.agda in 
    BFLib-0.1.  Is not it natural?

Reply:
I agree that "unsigned natural" and "signed natural" don't make sense.

But we're not talking about `Data.Natural.(Signed/Unsigned)`. We are
talking about `Data.Bin.(Signed/Unsigned)` which correspond to `Data.Nat`
and `Data.Integer`.

Reply:
> This is an algebraic part of the library.

What do you mean by this? It lives in the `Data` folder...

> See the declaration of a binary represented integer ℤb in BInteger/BI0.agda in
BFLib-0.1. Does not it look naturally?

To me a binary integer means an integer that is either 0 or 1.

Reply:
> To me a binary integer means an integer that is either 0 or 1.

And Bin also?

@gallais > 
gallais 4 hours ago •
> I agree that "unsigned natural" and "signed natural" don't make sense.
> But we're not talking about Data.Natural.(Signed/Unsigned). We are
> talking about Data.Bin.(Signed/Unsigned) which correspond to Data.Nat
> and Data.Integer.

Well, in lib-1.0 we have Data/Integer/ and Data/Bin/.
In the proposed lib  we could have Data/Integer, Data/Bin/,  Data/Bin/Integer/.
``Data/Bin/*.agda``  are for the Bin data,
``Data/Bin/Integer/*.agda``  to be for the  ``ℤb`` data (binary represented integer).
?

Reply:
@MatthewDaggitt  wrote

>>  This is an algebraic part of the library.

> What do you mean by this? It lives in the Data folder...

A _notion_ is more important than its Data representation. There are possible many data representations for the same notion. 
For example, ``ℤ`` denotes mainly a certain notion, the _laws_ of integer numbers. This is of _algebra_. And the Data folder and a representation "a natural + sign" -- these are technical  details needed for particular algorithms.
I do not claim that algorithms need to ignore data structures.  
But in the first place, ``ℤ`` denotes a notion, and its _data_ structure is on the second place.